src/abbfreeathome/devices/room_temperature_controller_master_without_fan.py

```diff
@@ -0,0 +1,189 @@
+"""Free@Home RoomTemperatureControllerMasterWithoutFan Class."""
+
+from typing import Any
+
+from ..api import FreeAtHomeApi
+from ..bin.pairing import Pairing
+from .base import Base
+
+
+class RoomTemperatureControllerMasterWithoutFan(Base):
+    """Free@Home RoomTemperatureControllerWithoutFan Class."""
+
+    _state_refresh_output_pairings: list[Pairing] = [
+        Pairing.AL_SET_POINT_TEMPERATURE,
+        Pairing.AL_STATE_INDICATION,
+        Pairing.AL_MEASURED_TEMPERATURE,
+        Pairing.AL_HEATING_DEMAND,
+        Pairing.AL_CONTROLLER_ON_OFF,
+    ]
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        channel_id: str,
+        channel_name: str,
+        inputs: dict[str, dict[str, Any]],
+        outputs: dict[str, dict[str, Any]],
+        parameters: dict[str, dict[str, Any]],
+        api: FreeAtHomeApi,
+        floor_name: str | None = None,
+        room_name: str | None = None,
+    ) -> None:
+        """Initialize the Free@Home RoomTemperatureControllerWithoutFan class."""
+        self._state: bool | None = None
+        self._current_temperature: float | None = None
+        self._valve: int | None = None
+        self._target_temperature: float | None = None
+        self._state_indication: int | None = None
+        self._eco_mode: bool | None = None
+
+        super().__init__(
+            device_id,
+            device_name,
+            channel_id,
+            channel_name,
+            inputs,
+            outputs,
+            parameters,
+            api,
+            floor_name,
+            room_name,
+        )
+
+    @property
+    def state(self) -> bool:
+        """Get the state of the RTC."""
+        return bool(self._state)
+
+    @property
+    def current_temperature(self) -> float:
+        """Get the current temperature."""
+        return float(self._current_temperature)
+
+    @property
+    def valve(self) -> int:
+        """Get the status of the valve."""
+        return int(self._valve)
```

Comment:
The initial value of these properties is `None`, if the RTC doesn’t have the output (for valve for example) then getting this property would fail as it can’t convert type `None` to `int`.

If you could convert the string value to `int()` in `_refresh_state_from_output` when setting the property, and allow this property to return `None` that’ll ensure we don’t get any conversion type errors if the output doesn’t exist.

Home Assistant can handle `None` types as well.

This goes for all properties, I’m just pointing out valve as an example.

```suggestion
    @property
    def valve(self) -> int | None:
        """Get the status of the valve."""
        return self._valve
```

See also: https://github.com/kingsleyadam/local-abbfreeathome/blob/main/src/abbfreeathome/devices/brightness_sensor.py

Reply:
Done